packages/babel-preset-react/src/normalize-options.js

```diff
@@ -0,0 +1,100 @@
+import { OptionValidator } from "@babel/helper-validator-option";
+const v = new OptionValidator("@babel/preset-react");
+
+export default function normalizeOptions(options = {}) {
+  if (process.env.BABEL_8_BREAKING) {
+    if ("useSpread" in options) {
+      throw new Error(
+        '@babel/preset-react: Since Babel 8, an inline object with spread elements is always used, and the "useSpread" option is no longer available. Please remove it from your config.',
+      );
+    }
+
+    if ("useBuiltIns" in options) {
+      const useBuiltInsFormatted = JSON.stringify(options.useBuiltIns);
+      throw new Error(
+        `@babel/preset-react: Since "useBuiltIns" is removed in Babel 8, you can remove it from the config.
+- Babel 8 now transforms JSX spread to object spread. If you need to transpile object spread with
+\`useBuiltIns: ${useBuiltInsFormatted}\`, you can use the following config
+{
+  "plugins": [
+    ["@babel/plugin-proposal-object-rest-spread", { "loose": true, "useBuiltIns": ${useBuiltInsFormatted} }]
+  ],
+  "presets": ["@babel/preset-react"]
+}`,
+      );
+    }
+
+    const TopLevelOptions = {
+      development: "development",
+      importSource: "importSource",
+      pragma: "pragma",
+      pragmaFrag: "pragmaFrag",
+      pure: "pure",
+      runtime: "runtime",
+      throwIfNamespace: "throwIfNamespace",
+    };
+    v.validateTopLevelOptions(options, TopLevelOptions);
+    const development = v.validateBooleanOption(
+      TopLevelOptions.development,
+      options.development,
+    );
+    const importSource = v.validateStringOption(
+      TopLevelOptions.importSource,
+      options.importSource,
+    );
+    const pragma = v.validateStringOption(
+      TopLevelOptions.pragma,
+      options.pragma,
+    );
+    const pragmaFrag = v.validateStringOption(
+      TopLevelOptions.pragmaFrag,
+      options.pragmaFrag,
+    );
+    const pure = v.validateBooleanOption(TopLevelOptions.pure, options.pure);
+    const runtime = v.validateStringOption(
+      TopLevelOptions.runtime,
+      options.runtime,
+      "automatic",
+    );
+    const throwIfNamespace = v.validateStringOption(
+      TopLevelOptions.throwIfNamespace,
+      options.throwIfNamespace,
+    );
+
+    return {
+      development,
+      importSource,
+      pragma,
+      pragmaFrag,
+      pure,
+      runtime,
+      throwIfNamespace,
+    };
+  } else {
+    let { pragma, pragmaFrag } = options;
```

Comment:
do we need to do it here too?
```
let { pragma, pragmaFrag, development = false } = opts;
```

kinda confused what below is doing.. maybe can just do`|| false`

```
const development = !!options.development
```

Reply:
I think it's different to keep the Babel 7 behavior the same (where we coerced everything to a boolean); if we use a default value in the destructuring it will change with falsy values.